source/Trap.hx

```diff
@@ -0,0 +1,70 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxObject;
+import flixel.FlxSprite;
+import flixel.tweens.FlxTween;
+import flixel.tweens.FlxEase;
+import flixel.system.FlxAssets.FlxGraphicAsset;
+import flixel.util.FlxColor;
+import flixel.input.keyboard.FlxKey;
+import flixel.group.FlxGroup;
+
+class Trap extends FlxSprite
+ {
+    private var subtrap1:Trap;
+    private var subtrap2:Trap;
+    private var subtrap3:Trap;
+
+    /**
+     * Intializer
+     *
+     * @param	X	Starting x coordinate
+     * @param	Y	Starting y coordinate
+     * @param	SimpleGraphic	Non-animating graphic. Nothing fancy (optional)
+     */
+    public function new(?X:Float=0, ?Y:Float=0, ?SimpleGraphic:FlxGraphicAsset)
+    {
+        super(X, Y, SimpleGraphic);
+
+        // Initializes a basic graphic for the Trap
+        makeGraphic(32, 32, FlxColor.RED);  
+
+    }
+
+    /**
+	 * Update function.
+	 *
+	 * Responsible for calling the update method of the current state and
+     * switching states if a new one is returned.
+	 *
+	 * @param	elapsed	Time passed since last call to update in seconds.
+	 */
+	public override function update(elapsed:Float):Void
+	{
+		super.update(elapsed);
+    
+	}
+
+
+    //Builds flame bar - adds tweens (WIP)
+    public function barCreate(trap1:Trap, trap2:Trap, trap3:Trap, x:Float, y:Float)
```

Comment:
You could use more comments describing what's going on in this class, in the style of the other comments above.